lib/utils/utf8.c

```diff
@@ -79,3 +80,45 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+int utf8_count_chars(const char *s)
+{
+	int count = 0;
+	const unsigned char *p = (const unsigned char *)s;
+
+	while (*p != '\0') {
+		if ((*p & 0x80) == 0) { /* 1-byte character: 0xxxxxxx */
+			p += 1;
+		} else if ((*p & 0xE0) == 0xC0) {
+			/* 2-byte character: 110xxxxx */
+			if ((p[1] & 0xC0) != 0x80) {
+				/* invalid continuation */
+				return -EINVAL;
+			}
+			p += 2;
+		} else if ((*p & 0xF0) == 0xE0) {
+			/* 3-byte character: 1110xxxx */
+			if ((p[1] & 0xC0) != 0x80
+				|| (p[2] & 0xC0) != 0x80) {
+				/* invalid continuation */
+				return -EINVAL;
+			}
+			p += 3;
+		} else if ((*p & 0xF8) == 0xF0) {
```

Comment:
We have the defines above: 

```c
#define SEQUENCE_FIRST_MASK 0xC0
#define SEQUENCE_LEN_2_BYTE 0xC0
#define SEQUENCE_LEN_3_BYTE 0xE0
#define SEQUENCE_LEN_4_BYTE 0xF0
```

I suggest that we use them (and maybe rename them if they don't fit the usage here with the current naming), rather than using magic numbers (e.g. 0xE0, etc.)